### PR TITLE
Remove Fedora 37 & add Fedora 39

### PR DESCRIPTION
**Pull request by nicrowe00**
Removing Fedora 37 from the CI as it's now EOL and adding Fedora 39.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,6 +32,8 @@ jobs:
         include:
           - container_image: registry.fedoraproject.org/fedora:38
             dotnet_version: "8.0"
+          - container_image: registry.fedoraproject.org/fedora:39
```
Comment (omajid) [inline .github/workflows/ci.yml:35]: .NET 8 should be available everywhere now, so maybe we can remove all these items under `include` and just add `8.0` to `dotnet_versions`?

Comment (nicrowe00) [reply .github/workflows/ci.yml]: > .NET 8 should be available everywhere now, so maybe we can remove all these items under `include` and just add `8.0` to `dotnet_versions`?

Sounds good, I've just updated it.

**Comment by omajid** [comment]
Oh. The CI config is trying to use `dnf` on alpine.

I think we can drop these lines now:

```
          if [[ ${{ matrix.dotnet_version }} == "8.0" ]]; then
            dnf install -y dnf-plugins-core
            if [[ ${{ matrix.container_image }} == *ubi8* ]] ; then
              dnf copr enable -y @dotnet-sig/dotnet-preview rhel-8-$(uname -m)
            else
              dnf copr enable -y @dotnet-sig/dotnet-preview
            fi
          fi
````

That was trying to use the https://copr.fedorainfracloud.org/coprs/g/dotnet-sig/dotnet-preview/ repo to install .NET 8, but .NET 8 should be available everywhere now.

**Comment by omajid** [comment]
@ayakael hey, do you know how we can install .NET 8 on alpine:edge? Based on https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/49831/diffs#c3a9fd6134014af4ee84828b0a9166e7f40f9d85, I was expecting the following to work, but it doesn't:

```
$ podman run -it alpine:edge
Resolved "alpine" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/alpine:edge...
Getting image source signatures
Copying blob b790c763077d done   | 
Copying config 23d4cb6bde done   | 
Writing manifest to image destination
...
/ # cat /etc/apk/repositories 
https://dl-cdn.alpinelinux.org/alpine/edge/main
https://dl-cdn.alpinelinux.org/alpine/edge/community
https://dl-cdn.alpinelinux.org/alpine/edge/testing
/ # apk update
fetch https://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/testing/x86_64/APKINDEX.tar.gz
v3.19.0-466-gc81aaeb1200 [https://dl-cdn.alpinelinux.org/alpine/edge/main]
v3.19.0-466-gc81aaeb1200 [https://dl-cdn.alpinelinux.org/alpine/edge/community]
v3.19.0-466-gc81aaeb1200 [https://dl-cdn.alpinelinux.org/alpine/edge/testing]
OK: 29721 distinct packages available
/ # apk add dotnet-sdk-8.0 dotnet-doc
ERROR: unable to select packages:
  dotnet-sdk-8.0 (no such package):
    required by: world[dotnet-sdk-8.0]
```

**Comment by omajid** [comment]
All errors are in tools-in-path and managed-symbols-are-available, both of which are known issues. I am merging this PR.

**Comment by ayakael** [comment]
@omajid Sorry for not having come back to you, december was a busy month. .NET 8 is now merged on edge. I've re-enabled CI via https://github.com/redhat-developer/dotnet-regular-tests/pull/330

**Pull request merged**